### PR TITLE
Use all cores instead of only 2 in CircleCI configuration

### PR DESCRIPTION
**Pull request by atilaneves**

### CODE REVIEW

File: .circleci/run.sh [hunk 1]
```diff
@@ -5,7 +5,7 @@ set -uexo pipefail
 HOST_DMD_VER=2.095.0
 CURL_USER_AGENT="CirleCI $(curl --version | head -n 1)"
 DUB=${DUB:-dub}
-N=${N:-2}
+N=${N:-$(getconf _NPROCESSORS_ONLN)}
```
Comment (ljmf00) [inline .circleci/run.sh:8]: Why not `nproc` ? While `getconf` is POSIX-compliant, `_NPROCESSORS_ONLN` is not specified and some BSD-based systems such as FreeBSD may not provide this. I think we should check for existence and fallback if necessary. For FreeBSD and similar, something like `sysctl hw.ncpu` is also a possibility.

Comment (atilaneves) [reply .circleci/run.sh:8]: I wasn't sure if this only ran on Linux.

Comment (ljmf00) [reply .circleci/run.sh:8]: Right, this is circleci specific, so doesn't really matter :)

**Comment by ljmf00** [comment]
OOM killer seem to be aggressive and kill some makefile targets with more parallel processes.

**Pull request closed without merging**